core/src/main/java/org/infinispan/eviction/EvictionManager.java

```diff
@@ -22,6 +23,17 @@
    /**
     * Handles notifications of evicted entries
     * @param evicted The entries that were just evicted
+    * @return stage that when complete the notifications are complete
     */
-   CompletionStage<Void> onEntryEviction(Map<K, Map.Entry<K, V>> evicted);
+   default CompletionStage<Void> onEntryEviction(Map<K, Map.Entry<K, V>> evicted) {
+      return onEntryEviction(evicted, null);
+   }
+
+   /**
+    * Handles notifications of evicted entries based on if the command allow them
+    * @param evicted The entries that were just evicted
+    * @param command The command that generated the eviction if applicable
+    * @return stage that when complete the notifications are complete
+    */
+   CompletionStage<Void> onEntryEviction(Map<K, Map.Entry<K, V>> evicted, FlagAffectedCommand command);
```

Comment:
Hrmm, I would like that. Didn't think of it since I was trying to keep it similar to the deprecated method. But I can change that default one to invoke this method N times instead. 👍